### PR TITLE
refactor(components): [option-group] switch to script-setup syntax

### PR DESCRIPTION
**Pull request by josonho**
refactor #6114 

***

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by JeremyWuuuuu** [comment]
please do rebase on the dev branch again for getting updated CI code @josonho

File: packages/components/select/index.ts [hunk 1]
```diff
@@ -13,3 +13,5 @@ export const ElOption = withNoopInstall(Option)
 export const ElOptionGroup = withNoopInstall(OptionGroup)
 
 export * from './src/token'
+
+export type OptionGroupInstance = InstanceType<typeof OptionGroup>
```
Comment (sxzz) [inline packages/components/select/index.ts:17]: move it to `option-group.ts`

File: packages/components/select/src/option-group.vue [hunk 2]
```diff
@@ -25,65 +24,58 @@ import {
...
+  disabled: {
+    type: Boolean,
+    default: false,
   },
-  setup(props) {
-    const ns = useNamespace('select')
-    const visible = ref(true)
-    const instance = getCurrentInstance()
-    const children = ref([])
+})
```
Comment (sxzz) [inline packages/components/select/src/option-group.vue:37]: Move props and emit to `option-group.ts` and export it.

File: packages/components/select/src/option-group.vue [hunk 3]
```diff
@@ -25,65 +24,58 @@ import {
...
+  selectGroupKey,
+  reactive({
+    ...toRefs(props),
+  })
+)
 
-    onMounted(() => {
-      children.value = flattedChildren(instance.subTree)
-    })
+const select = inject(selectKey)
```
Comment (sxzz) [inline packages/components/select/src/option-group.vue:51]:
```suggestion
const select = inject(selectKey)!
```

File: packages/components/select/src/option-group.vue [hunk 4]
```diff
@@ -25,65 +24,58 @@ import {
...
+  }
+  return children
+}
 
-    return {
-      visible,
-      ns,
-    }
-  },
+const { groupQueryChange } = toRaw(select)!
```
Comment (sxzz) [inline packages/components/select/src/option-group.vue:77]:
```suggestion
const { groupQueryChange } = toRaw(select)
```

**Review by sxzz — COMMENTED** [review]
Please rebase your code, and check TS error again.

**Comment by xiaoxian521** [comment]
Hey, thank you very much for your contribution, but here are some conflicts that need to be solved by you. We can continue to review the code after solving. One thing to remind is that if the conflicts is still not resolved after `10` days, sorry, we will close this pr ❤️

**Pull request closed without merging**